Use strict comparision and compare with section values for tab section screen ID

moon0326 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #55261  .

This PR fixes a bug where tab sections reported their parent screen ID incorrectly.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Install and activate  [woo-page-screen-id-debugger.zip](https://github.com/user-attachments/files/18727804/woo-page-screen-id-debugger.zip)
3. Navigate to WooCommerce -> Settings -> Products or any other pages with tab sections.
4. Click one of its tab sections.
5. You should see the `Current Screen ID` value with the section I.D you're currently on.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

This PR fixes a bug where the tab sections reported their parent screen ID incorrectly in WooCommerce settings.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

Commit:
Use strict comparision and compare with section values

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

chihsuan reviewed (commented):
@moon0326 Thanks for the fix!

> Install and activate https://gist.github.com/moon0326/e6f5d509bf8168e3ad2faf9c7a549dc5 

I got this error: `Non-static method Automattic\WooCommerce\Admin\PageController::get_current_screen_id() cannot be called statically` when I try to activate the plugin.

Here is the fixed plugin zip file: [woo-page-screen-id-debugger.zip](https://github.com/user-attachments/files/18727804/woo-page-screen-id-debugger.zip)


I tested a few pages, most of them have the correct screen ID. But for some pages, it shows the parent screen ID. For example, `WooCommerce -> Settings -> Advanced -> Features`, the screen ID `woocommerce_page_wc-settings-advanced` but it should be `woocommerce_page_wc-settings-advanced-features`. Is this expected?

<img width="1023" alt="Screenshot 2025-02-10 at 11 32 08" src="https://github.com/user-attachments/assets/1cecf3b7-7efa-4d61-afb7-7c392b299e7b" />

Commit:
Add blueprint and features

moon0326 commented:
Thank you @chihsuan 👍 

> Here is the fixed plugin zip file: [woo-page-screen-id-debugger.zip](https://github.com/user-attachments/files/18727804/woo-page-screen-id-debugger.zip)

🙇 I've updated the test instructions.

> I tested a few pages, most of them have the correct screen ID. But for some pages, it shows the parent screen ID. For example, WooCommerce -> Settings -> Advanced -> Features, the screen ID woocommerce_page_wc-settings-advanced but it should be woocommerce_page_wc-settings-advanced-features. Is this expected?

Good catch. "features" was not listed. I've added both features and blueprint in [Add blueprint and features](https://github.com/woocommerce/woocommerce/pull/55262/commits/16f38e6af8d3441ecbbc15175d3d5cee8c79a278)

Commit:
Merge branch 'trunk' into fix/55261-incorrect-tab-section-screen-id

chihsuan reviewed (approved):
> Good catch! "features" was not listed. I've added both features and blueprint in [Add blueprint and features](https://github.com/woocommerce/woocommerce/pull/55262/commits/16f38e6af8d3441ecbbc15175d3d5cee8c79a278)

Thanks @moon0326, I confirmed that the screen ID is correct for features now. 

I found some other tabs also not listed:

- `WooCommerce > Settings > Products > Approved download directories`
- `WooCommerce > Settings > Products > Advanced`
- `WooCommerce > Settings > Shipping > Local pickup`

I'm not sure if it's possible to automatically add all the tabs to the list by looping through the settings. If that works, maybe we should do that.

I'll pre-approve this PR.

Commit:
Add a few more sections

moon0326 commented:
> > Good catch! "features" was not listed. I've added both features and blueprint in [Add blueprint and features](https://github.com/woocommerce/woocommerce/pull/55262/commits/16f38e6af8d3441ecbbc15175d3d5cee8c79a278)
> 
> Thanks @moon0326, I confirmed that the screen ID is correct for features now.
> 
> I found some other tabs also not listed:
> 
> * `WooCommerce > Settings > Products > Approved download directories`
> * `WooCommerce > Settings > Products > Advanced`
> * `WooCommerce > Settings > Shipping > Local pickup`
> 
> I'm not sure if it's possible to automatically add all the tabs to the list by looping through the settings. If that works, maybe we should do that.
> 
> I'll pre-approve this PR.

Thank you! I've added them in [Add a few more sections](https://github.com/woocommerce/woocommerce/pull/55262/commits/e4133fb0a802a8fe5720faa6539e6a03aca2412c)

> I'm not sure if it's possible to automatically add all the tabs to the list by looping through the settings. If that works, maybe we should do that.

I think it's possible. Let's work on that in a separate PR. I'll create an issue.

Commit:
Merge branch 'trunk' into fix/55261-incorrect-tab-section-screen-id
